packages/cardano-services/test/Asset/AssetHttpService.test.ts

```diff
@@ -0,0 +1,125 @@
+import { AssetHttpService, HttpServer, HttpServerConfig } from '../../src';
+import { Pool } from 'pg';
+import { fromSerializableObject } from '@cardano-sdk/util';
+import { getPort } from 'get-port-please';
+import { dummyLogger as logger } from 'ts-log';
+import axios from 'axios';
+
+const APPLICATION_JSON = 'application/json';
+const APPLICATION_CBOR = 'application/cbor';
+const UNSUPPORTED_MEDIA_STRING = 'Request failed with status code 415';
+const BAD_REQUEST_STRING = 'Request failed with status code 400';
+
+describe('AssetHttpService', () => {
+  let apiUrlBase: string;
+  let config: HttpServerConfig;
+  let db: Pool;
+  let httpServer: HttpServer;
+  let service: AssetHttpService;
+  let port: number;
+
+  beforeAll(async () => {
+    port = await getPort();
+    apiUrlBase = `http://localhost:${port}/asset`;
+    config = { listen: { port } };
+  });
+
+  describe('healthy state', () => {
+    beforeAll(async () => {
+      db = new Pool({ connectionString: process.env.DB_CONNECTION_STRING });
+      service = new AssetHttpService({ db, logger });
+      httpServer = new HttpServer(config, { services: [service] });
+      await httpServer.initialize();
+      await httpServer.start();
+    });
+
+    afterAll(async () => {
+      await httpServer.shutdown();
+      await db.end();
+    });
+
+    describe('/health', () => {
+      it('/health response should be true', async () => {
+        const res = await axios.post(`${apiUrlBase}/health`, undefined, {
+          headers: { 'Content-Type': APPLICATION_JSON }
+        });
+        expect(res.status).toBe(200);
+        expect(res.data).toEqual({ ok: true });
+      });
+    });
+
+    describe('/get-asset', () => {
+      it('returns a 415 coded response if the wrong content type header is used', async () => {
+        try {
+          await axios.post(`${apiUrlBase}/get-asset`, { args: [] }, { headers: { 'Content-Type': APPLICATION_CBOR } });
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        } catch (error: any) {
+          expect(error.response.status).toBe(415);
+          expect(error.message).toBe(UNSUPPORTED_MEDIA_STRING);
+        }
+      });
+
+      it('returns 400 coded response if the request is bad formed', async () => {
+        try {
+          await axios.post(`${apiUrlBase}/get-asset`, { args: [['test']] });
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        } catch (error: any) {
+          expect(error.response.status).toBe(400);
+          expect(error.message).toBe(BAD_REQUEST_STRING);
+        }
+      });
+
+      it('internal server error for not existing existing asset id', async () => {
```

Comment:
I don't think this is the expected behavior...

Reply:
Nope, it should be ProviderError{ProviderFailure.NotFound}

Reply:
Let's go on in [this discussion](https://github.com/input-output-hk/cardano-js-sdk/pull/321#discussion_r924385192) as it is the same topic.